### ISSUE TITLE
Support miniumum and maxiumum run time

### ISSUE THREAD

**Issue by alistair23** [InventoCasa/ha-advanced-blueprints#32]
Pool pumps and other devices have a minimum and maximum number of hours they need to run in a day.

That is even if there is no sun the device must run for X (minimum) number of hours and even if there is excess sun it doesn't need to run more then Y (maximum) number of hours. 

It would be great to support optional minimum and maximum run times.

I'm thinking it could be doable by implementing a run counter when devices are turned [on](https://github.com/InventoCasa/ha-advanced-blueprints/blob/main/PV_Excess_Control/pyscript/pv_excess_control.py#L319) and [off](https://github.com/InventoCasa/ha-advanced-blueprints/blob/main/PV_Excess_Control/pyscript/pv_excess_control.py#L381). The just adding a check for maximum runtime [before starting the device](https://github.com/InventoCasa/ha-advanced-blueprints/blob/main/PV_Excess_Control/pyscript/pv_excess_control.py#L318). Then the minimum could be supported by adding another loop in the [trigger factory](https://github.com/InventoCasa/ha-advanced-blueprints/blob/main/PV_Excess_Control/pyscript/pv_excess_control.py#L398) that kicks in after a specified time (the time when electricity becomes cheaper)

**Comment by Kolbi** [comment]
I would also like to have such a feature.

**Comment by alistair23** [comment]
I have a fork that supports min/max runtime: https://github.com/alistair23/ha-advanced-blueprints/tree/alistair/min-max-runtime

I'm working on upstreaming changes, but this repo seems to be unmaintained

**Comment by InventoCasa** [comment]
Hi there, the repo ist still maintained but I don't have time to look into new features/ PRs at the moment. This may get better in the colder months this year (from approx. October on). Till then I can only spare time to fix bugs, in case there are any.

**Comment by Kolbi** [comment]
If maybe @alistair23 provides a PR, @InventoCasa can merge it maybe?

Otherwise @alistair23 I will try your repo, till @InventoCasa has time to update his :)

**Comment by GeorgeBark** [comment]
@alistair23 is your fork working properly? I am also interested in this, not sure if you are still developing and testing or if I can already use yours. Also, I see a cron for 20:00 to check the time, can this be configurable? as in, check if it hasn't reached the minimum at 3am instead of at 8pm.

Thanks!

**Comment by alistair23** [comment]
Yes, it is. I'm using it every day and it is actively tested and developed.

I have added a few useful features related to min and max.

You can easily change the time in the script. I'm trying to remove all of the configuration options as it makes it difficult to have multiple automations. But 3AM seems pretty late. You only have a few hours to meet the minimum run time before the sun rises again.

**Comment by mzst123** [comment]
Hi, I am also interested in this.
It would also be nice to have a summer/winter schedule, because in my situation for example, in the colder months, the pool does not need to run for as many hours as during summer. Is this easy to do?

Another question, to update, do I just copy the new files over or do I have to redo the automation completely?
Thank you

**Comment by InventoCasa** [comment]
Will integrate this into my next release of the script. Thanks for contributing @alistair23

**Comment by alistair23** [comment]
> It would also be nice to have a summer/winter schedule, because in my situation for example, in the colder months, the pool does not need to run for as many hours as during summer. Is this easy to do?

It's not that hard to do, but figuring out the colder months and having a UI to set the different times is a bit clunky



> Another question, to update, do I just copy the new files over or do I have to redo the automation completely?

Both, you will need to copy the files over and then update the automation to get the new settings

**Comment by alistair23** [comment]
> Will integrate this into my next release of the script. Thanks for contributing @alistair23

The PR is open, so just merge it whenever

**Comment by nicorusti** [comment]
+1 for this feature, would be indeed helpful for pumps!
